sunpy/map/mapbase.py

```diff
@@ -1573,12 +1573,12 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
 
         pixel_array_center = (np.flipud(new_data.shape) - 1) / 2.0
 
-        # Create a temporary map so we can use it for the data to pixel calculation.
-        temp_map = self._new_instance(new_data, new_meta, self.plot_settings)
+        # Create a temporary WCS so we can use it for the data to pixel calculation.
+        new_wcs = self.wcs.deepcopy()
+        new_wcs.wcs.crpix += [pad_x, pad_y]
```

Comment:
```suggestion
        new_wcs = WCS(new_meta)
```
Since the padding is already added to `new_meta` above, this avoids having to add it again to the WCS?

Reply:
Nope, that doesn't work because creating the `WCS` instance that way doesn't inject our logic that isn't added to the metadata (e.g., assumptions about observation time and observer location).  Regardless, it turns out this WCS calculation isn't needed.